<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>06-binormal-model.utf8</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown #bookdown:version# and GitBook 2.6.7" />

  <meta property="og:title" content="06-binormal-model.utf8" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="06-binormal-model.utf8" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<!--bookdown:link_prev-->
<!--bookdown:link_next-->
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



<!--bookdown:title:start-->
<!--bookdown:title:end-->

<!--bookdown:toc:start-->
  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">
<!--bookdown:toc2:start-->
<ul>
<li><a href="#BinMod"><span class="toc-section-number">1</span> Binormal model</a>
<ul>
<li><a href="#BinModIntro"><span class="toc-section-number">1.1</span> Introduction</a></li>
<li><a href="#BinModTheModel"><span class="toc-section-number">1.2</span> The binormal model</a>
<ul>
<li><a href="#binning-the-data"><span class="toc-section-number">1.2.1</span> Binning the data</a></li>
<li><a href="#invariance-of-the-binormal-model-to-arbitrary-monotone-transformations"><span class="toc-section-number">1.2.2</span> Invariance of the binormal model to arbitrary monotone transformations</a></li>
<li><a href="#expressions-for-sensitivity-and-specificity"><span class="toc-section-number">1.2.3</span> Expressions for sensitivity and specificity</a></li>
<li><a href="#binormal-model-in-standard-notation"><span class="toc-section-number">1.2.4</span> Binormal model in standard notation</a></li>
<li><a href="#properties-of-the-binormal-model-roc-curve"><span class="toc-section-number">1.2.5</span> Properties of the binormal model ROC curve</a></li>
<li><a href="#density-functions-pdfs-of-the-binormal-model"><span class="toc-section-number">1.2.6</span> Density functions (pdfs) of the binormal model</a></li>
</ul></li>
<li><a href="#fitting-an-roc-curve-to-data-points"><span class="toc-section-number">1.3</span> Fitting an ROC curve to data points</a>
<ul>
<li><a href="#a-java-fitted-roc-curve"><span class="toc-section-number">1.3.1</span> A JAVA fitted ROC curve</a></li>
<li><a href="#a-simplistic-straight-line-fit-to-the-roc-curve"><span class="toc-section-number">1.3.2</span> A simplistic straight line fit to the ROC curve</a></li>
<li><a href="#maximum-likelihood-estimation-mle"><span class="toc-section-number">1.3.3</span> Maximum likelihood estimation (MLE)</a></li>
<li><a href="#code-implementing-mle"><span class="toc-section-number">1.3.4</span> Code implementing MLE</a></li>
<li><a href="#validating-the-fitting-model"><span class="toc-section-number">1.3.5</span> Validating the fitting model</a></li>
<li><a href="#estimating-the-covariance-matrix"><span class="toc-section-number">1.3.6</span> Estimating the covariance matrix</a></li>
<li><a href="#estimating-the-variance-of-az"><span class="toc-section-number">1.3.7</span> Estimating the variance of Az</a></li>
<li><a href="#single-fom-derived-from-roc-curve"><span class="toc-section-number">1.3.8</span> Single FOM derived from ROC curve</a></li>
</ul></li>
<li><a href="#BinMod-Discussion"><span class="toc-section-number">1.4</span> Discussion</a></li>
<li><a href="#BinMod-references"><span class="toc-section-number">1.5</span> References</a></li>
</ul></li>
</ul>
<!--bookdown:toc2:end-->
      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<!--bookdown:toc:end-->
<!--bookdown:body:start-->
<div id="BinMod" class="section level1" number="1">
<h1 number="1"><span class="header-section-number">1</span> Binormal model</h1>
<div id="BinModIntro" class="section level2" number="1.1">
<h2 number="1.1"><span class="header-section-number">1.1</span> Introduction</h2>
<p>The equal variance binormal model was described in Chapter 02. The ratings method of acquiring ROC data and calculation of operating points was discussed in Chapter 04. It was shown, Fig. @ref(fig:ratingsParadigmEqVarFitA), that for a clinical dataset the unequal-variance binormal model visually fitted the data better than the equal-variance binormal model, although how the unequal variance fit was obtained was not discussed. This chapter deals with details of the unequal-variance binormal model, often abbreviated to <strong>binormal model</strong>, establishes necessary notation, and derives expressions for sensitivity, specificity and the area under the predicted ROC curve).</p>
<p>The binormal model describes univariate datasets, in which there is one ROC rating per case, as in a single observer interpreting cases, one at a time, in a single modality. By convention the qualifier “univariate” is often omitted. In TBA Chapter 21 a bivariate model will be described where each case yields two ratings, as in a single observer interpreting cases in two modalities, or the homologous problem of two observers interpreting cases in a single modality.</p>
<p>The main aim of this chapter is to demystify statistical curve fitting. With the passing of Dorfman, Metz and Swensson, parametric modeling is being neglected. Researchers are instead focusing on non-parametric analysis using the empirical AUC. While useful, empirical AUC yields almost no insight into what is limiting performance. Taking the mystery out of curve fitting will allow the reader to appreciate later chapters that describe more complex fitting methods, which yield important insights into factors limiting performance.</p>
<p>Here is the organization of this chapter. It starts with a description of the binormal model and how it accommodates data binning. An important point, on which there is much confusion, on the invariance of the binormal model to arbitrary monotone transformations of the ratings is explicated with an example. Expressions for sensitivity and specificity are derived. Two notations used to characterize the binormal model are explained. Expressions for the pdfs of the binormal model are derived. A simple linear fitting method is illustrated: this used to be the only recourse a researcher had before Dorfman and Alf’s seminal publication <span class="citation">[@RN1081]</span>. The maximum likelihood method for estimating parameters of the binormal model is detailed. Validation of the fitting method is described, i.e., how can one be confident that the fitting method, which makes normality and other assumptions, is valid for a dataset arising from an unknown distribution. The Appendix has a detailed derivation, originally published in a terse paper <span class="citation">[@RN2342]</span> on the partial area under the ROC curve. The partial area is defined by the area under the binormal ROC curve from <span class="math inline">\(FPF = 0\)</span> to <span class="math inline">\(FPF = c\)</span>, where <span class="math inline">\(0 \leq c \leq 1\)</span>. As a special case <span class="math inline">\(c = 1\)</span> yields the total area under the binormal ROC.</p>
</div>
<div id="BinModTheModel" class="section level2" number="1.2">
<h2 number="1.2"><span class="header-section-number">1.2</span> The binormal model</h2>
<p>The unequal-variance binormal model (henceforth abbreviated to binormal model; when the author means equal variances, it will be made explicit) is defined by (capital letters indicate random variables and their lower-case counterparts are actual realized values):</p>
<p><span class="math display">\[\begin{equation} 
Z_{k_tt}\sim N\left ( \mu_t,\sigma_{t}^{2} \right );t=1,2
(\#eq:BinModZSampling1)
\end{equation}\]</span></p>
<p>where</p>
<p><span class="math display">\[\begin{equation} 
\mu_1=0;\mu_2=\mu;\sigma_{1}^{2}=1;\sigma_{2}^{2}=\sigma^{2}
(\#eq:BinModZSampling2)
\end{equation}\]</span></p>
<p>Eqn. @ref(eq:BinModZSampling1) states that the z-samples for non-diseased cases are distributed as a <span class="math inline">\(N(0,1)\)</span> distribution, i.e., the unit normal distribution, while the z-samples for diseased cases are distributed as a <span class="math inline">\(N(\mu,\sigma^2)\)</span> distribution, i.e., a normal distribution with mean <span class="math inline">\(\mu\)</span> and variance <span class="math inline">\(\sigma^2\)</span>. This is a 2-parameter model of the z-samples, not counting additional threshold parameters needed for data binning.</p>
<p>A more complicated version of this model allows the mean of the non-diseased distribution to be non-zero and its variance different from unity. The 4-parameter model is no more general than the 2-parameter model. The reason is that one is free to transform the decision variable, and associated thresholds, by applying arbitrary monotonic increasing function transformation, which do not change the ordering of the ratings and hence do not change the ROC curve. So if the mean of the noise distribution were non-zero, subtracting this value from all Z-samples would shift the effective mean of the non-diseased distribution to zero (the shifted Z-values are monotonically related to the original values) and the mean of the shifted diseased distribution becomes <span class="math inline">\(\mu_2-\mu_1\)</span>. Next, one scales or divides (division by a positive number is also a monotonic transformation) all the Z-samples by <span class="math inline">\(\sigma_1\)</span>, resulting in the scaled non-diseased distribution having unit variance, and the scaled diseased distribution has mean <span class="math inline">\(\frac{\mu_2-\mu_1}{\sigma_1}\)</span> and variance <span class="math inline">\((\frac{\sigma_2}{\sigma_1})^2\)</span>. Therefore, if one starts with 4 parameters then one can, by simple shifting and scaling operations, reduce the model to 2 parameters, as in Eqn. @ref(eq:BinModZSampling1). [The author has seen a publication on Bayesian ROC estimation using the four-parameter model.]</p>
<div id="binning-the-data" class="section level3" number="1.2.1">
<h3 number="1.2.1"><span class="header-section-number">1.2.1</span> Binning the data</h3>
<p>In an R-rating ROC study the observed ratings r take on integer values, 1 through R, it being understood that higher ratings correspond to greater confidence for disease. Defining dummy cutoffs <span class="math inline">\(\zeta_0 = -\infty\)</span> and <span class="math inline">\(\zeta_R = +\infty\)</span>, the binning rule for a case with realized z-sample z is (Chapter @ref(ratingsParadigm), Eqn. @ref(eq:ratingsParadigm-binningRule)):</p>
<p><span class="math display">\[\begin{equation} 
\text{if} \left (\zeta_{r-1} \le z \le \zeta_r  \right )\Rightarrow \text {rating} = r
(\#eq:BinModZBinning)
\end{equation}\]</span></p>
<p><img src="06-binormal-model_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<p>In the unequal-variance binormal model, the variance <span class="math inline">\(\sigma^2\)</span> of the z-samples for diseased cases is allowed to be different from unity. Most ROC datasets are consistent with <span class="math inline">\(\sigma &gt; 1\)</span>. The above figure, generated with <span class="math inline">\(\mu = 1.5, \sigma = 1.5, \zeta_1 = -2, \zeta_2 = -0.5, \zeta_3 = 1, \zeta_4 = 2.5\)</span>, illustrates how realized z-samples are converted to ratings, i.e., application of the binning rule @ref(eq:BinModZBinning). For example, a case with z-sample equal to -2.5 would be rated “1”, and one with z-sample equal to -1 would be rated “2”, cases with z-samples greater than 2.5 would be rated “5”, etc.</p>
</div>
<div id="invariance-of-the-binormal-model-to-arbitrary-monotone-transformations" class="section level3" number="1.2.2">
<h3 number="1.2.2"><span class="header-section-number">1.2.2</span> Invariance of the binormal model to arbitrary monotone transformations</h3>
<p>The binormal model is not as restrictive as might appear at first sight. Any monotone increasing transformation <span class="math inline">\(Y=f(Z)\)</span> applied to the observed z-samples, and the associated thresholds, will yield the same observed data, e.g., Table @ref(tab:ratingsParadigmExampleTable). This is because such a transformation leaves the ordering of the ratings unaltered and hence results in the same operating points. While the distributions for <span class="math inline">\(Y\)</span> will not be binormal (i.e., two independent normal distributions), one can safely “pretend” that one is still dealing with an underlying binormal model. An alternative way of stating this is that any pair of distributions is allowed as long as they are reducible to a binormal model form by a monotonic increasing transformation of Y: e.g., <span class="math inline">\(Z=f^{-1}\)</span>. [If <span class="math inline">\(f\)</span> is a monotone increasing function of its argument, so is <span class="math inline">\(f^{-1}\)</span>}.] For this reason, the term “pair of latent underlying normal distributions” is sometimes used to describe the binormal model. The robustness of the binormal model has been investigated <span class="citation">[@RN1216; @RN100]</span>. The referenced paper by Dorfman et al has an excellent discussion of the robustness of the binormal model.</p>
<p>The robustness of the binormal model, i.e., the flexibility allowed by the infinite choices of monotonic increasing functions, application of each of which leaves the ordering of the data unaltered, is widely misunderstood. The non-Gaussian appearance of histograms of ratings in ROC studies can lead one to incorrect conclusions that the binormal model is inapplicable to these datasets. To quote a reviewer of one of the author’s recent papers:</p>
<blockquote>
<p>I have had multiple encounters with statisticians who do not understand this difference…. They show me histograms of data, and tell me that the data is obviously not normal, therefore the binormal model should not be used.</p>
</blockquote>
<p>The reviewer is correct. The misconception is illustrated next.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># shows that monotone transformations have no effect on </span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="co"># AUC even though the pdfs look non-gaussian</span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="co"># common statistician misconception about ROC analysis</span></span>
<span id="cb1-4"><a href="#cb1-4"></a>fArray &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>)</span>
<span id="cb1-5"><a href="#cb1-5"></a>seedArray &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">11</span>,<span class="dv">12</span>)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="cf">for</span> (row <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>) { </span>
<span id="cb1-7"><a href="#cb1-7"></a>  f &lt;-<span class="st"> </span>fArray[row]</span>
<span id="cb1-8"><a href="#cb1-8"></a>  seed &lt;-<span class="st"> </span>seedArray[row]</span>
<span id="cb1-9"><a href="#cb1-9"></a>  <span class="kw">set.seed</span>(seed) </span>
<span id="cb1-10"><a href="#cb1-10"></a>  <span class="co"># numbers of cases simulated</span></span>
<span id="cb1-11"><a href="#cb1-11"></a>  K1 &lt;-<span class="st"> </span><span class="dv">900</span></span>
<span id="cb1-12"><a href="#cb1-12"></a>  K2 &lt;-<span class="st"> </span><span class="dv">1000</span></span>
<span id="cb1-13"><a href="#cb1-13"></a>  mu1 &lt;-<span class="st"> </span><span class="dv">30</span></span>
<span id="cb1-14"><a href="#cb1-14"></a>  sigma1 &lt;-<span class="st"> </span><span class="dv">7</span></span>
<span id="cb1-15"><a href="#cb1-15"></a>  mu2 &lt;-<span class="st"> </span><span class="dv">55</span></span>
<span id="cb1-16"><a href="#cb1-16"></a>  sigma2 &lt;-<span class="st"> </span><span class="dv">7</span> </span>
<span id="cb1-17"><a href="#cb1-17"></a>  <span class="co"># Simulate true gaussian ratings using above parameter values</span></span>
<span id="cb1-18"><a href="#cb1-18"></a>  z1 &lt;-<span class="st"> </span><span class="kw">rnorm</span>(K1,<span class="dt">mean =</span> mu1,<span class="dt">sd =</span> sigma1)</span>
<span id="cb1-19"><a href="#cb1-19"></a>  z1[z1<span class="op">&gt;</span><span class="dv">100</span>] &lt;-<span class="st"> </span><span class="dv">100</span>;z1[z1<span class="op">&lt;</span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="dv">0</span> <span class="co"># constrain to 0 to 100</span></span>
<span id="cb1-20"><a href="#cb1-20"></a>  z2 &lt;-<span class="st"> </span><span class="kw">rnorm</span>(K2,<span class="dt">mean =</span> mu2,<span class="dt">sd =</span> sigma2)</span>
<span id="cb1-21"><a href="#cb1-21"></a>  z2[z2<span class="op">&gt;</span><span class="dv">100</span>] &lt;-<span class="st"> </span><span class="dv">100</span>;z2[z2<span class="op">&lt;</span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="dv">0</span> <span class="co"># constrain to 0 to 100</span></span>
<span id="cb1-22"><a href="#cb1-22"></a>  <span class="co"># calculate AUC for true Gaussian ratings</span></span>
<span id="cb1-23"><a href="#cb1-23"></a>  AUC1 &lt;-<span class="st"> </span><span class="kw">TrapezoidalArea</span>(z1, z2)</span>
<span id="cb1-24"><a href="#cb1-24"></a>  Gaussians &lt;-<span class="st"> </span><span class="kw">c</span>(z1, z2)</span>
<span id="cb1-25"><a href="#cb1-25"></a>  <span class="co"># display histograms of true Gaussian ratings, A1, A2 or A3</span></span>
<span id="cb1-26"><a href="#cb1-26"></a>  x &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">x=</span>Gaussians) <span class="co">#  line 27</span></span>
<span id="cb1-27"><a href="#cb1-27"></a>  x &lt;-<span class="st">  </span></span>
<span id="cb1-28"><a href="#cb1-28"></a><span class="st">    </span><span class="kw">ggplot</span>(<span class="dt">data =</span> x, <span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> x)) <span class="op">+</span></span>
<span id="cb1-29"><a href="#cb1-29"></a><span class="st">    </span><span class="kw">geom_histogram</span>(<span class="dt">binwidth =</span> <span class="dv">5</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">fill=</span><span class="st">&quot;grey&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb1-30"><a href="#cb1-30"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="dt">label =</span> <span class="st">&quot;Original Rating&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb1-31"><a href="#cb1-31"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="dt">label =</span> <span class="kw">paste0</span>(<span class="st">&quot;A&quot;</span>, row, <span class="st">&quot;: &quot;</span>, <span class="st">&quot;Gaussians&quot;</span>))</span>
<span id="cb1-32"><a href="#cb1-32"></a>  <span class="kw">print</span>(x)</span>
<span id="cb1-33"><a href="#cb1-33"></a>  z &lt;-<span class="st"> </span><span class="kw">seq</span>(<span class="fl">0.0</span>, <span class="dv">100</span>, <span class="fl">0.1</span>)</span>
<span id="cb1-34"><a href="#cb1-34"></a>  <span class="co"># transform the latent Gaussians to true Gaussians</span></span>
<span id="cb1-35"><a href="#cb1-35"></a>  transformation &lt;-<span class="st"> </span></span>
<span id="cb1-36"><a href="#cb1-36"></a><span class="st">    </span><span class="kw">data.frame</span>(</span>
<span id="cb1-37"><a href="#cb1-37"></a>      <span class="dt">x =</span> z, </span>
<span id="cb1-38"><a href="#cb1-38"></a>      <span class="dt">z =</span>  <span class="kw">Y</span>(z,mu1,mu2,sigma1,sigma2,f))</span>
<span id="cb1-39"><a href="#cb1-39"></a>  <span class="co"># display transformation functions, B1, B2 or B3</span></span>
<span id="cb1-40"><a href="#cb1-40"></a>  x &lt;-<span class="st"> </span></span>
<span id="cb1-41"><a href="#cb1-41"></a><span class="st">    </span><span class="kw">ggplot</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> z)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb1-42"><a href="#cb1-42"></a><span class="st">    </span><span class="kw">geom_line</span>(<span class="dt">data =</span> transformation, <span class="dt">size =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb1-43"><a href="#cb1-43"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="dt">label =</span> <span class="st">&quot;Original Rating&quot;</span>) <span class="op">+</span></span>
<span id="cb1-44"><a href="#cb1-44"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="dt">label =</span> <span class="st">&quot;Transformed Rating&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb1-45"><a href="#cb1-45"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="dt">label =</span> <span class="kw">paste0</span>(<span class="st">&quot;B&quot;</span>, row, <span class="st">&quot;: &quot;</span>,<span class="st">&quot;Monotone Transformation&quot;</span>))</span>
<span id="cb1-46"><a href="#cb1-46"></a>  <span class="kw">print</span>(x)</span>
<span id="cb1-47"><a href="#cb1-47"></a>  y &lt;-<span class="st"> </span><span class="kw">Y</span>(<span class="kw">c</span>(z1, z2),mu1,mu2,sigma1,sigma2,f)</span>
<span id="cb1-48"><a href="#cb1-48"></a>  y1 &lt;-<span class="st"> </span>y[<span class="dv">1</span><span class="op">:</span>K1];y2 &lt;-<span class="st"> </span>y[(K1<span class="op">+</span><span class="dv">1</span>)<span class="op">:</span>(K1<span class="op">+</span>K2)]</span>
<span id="cb1-49"><a href="#cb1-49"></a>  <span class="co"># calculate AUC for transformed ratings</span></span>
<span id="cb1-50"><a href="#cb1-50"></a>  AUC2 &lt;-<span class="st"> </span><span class="kw">TrapezoidalArea</span>( y1, y2)</span>
<span id="cb1-51"><a href="#cb1-51"></a>  <span class="co"># display histograms of latent Gaussian ratings, C1, C2 or C3</span></span>
<span id="cb1-52"><a href="#cb1-52"></a>  x &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">x=</span>y)</span>
<span id="cb1-53"><a href="#cb1-53"></a>  x &lt;-<span class="st">  </span><span class="kw">ggplot</span>(<span class="dt">data =</span> x, <span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> x)) <span class="op">+</span></span>
<span id="cb1-54"><a href="#cb1-54"></a><span class="st">    </span><span class="kw">geom_histogram</span>(<span class="dt">binwidth =</span> <span class="dv">5</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">fill=</span><span class="st">&quot;grey&quot;</span>) <span class="op">+</span></span>
<span id="cb1-55"><a href="#cb1-55"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="dt">label =</span> <span class="st">&quot;Transformed Rating&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb1-56"><a href="#cb1-56"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="dt">label =</span> <span class="kw">paste0</span>(<span class="st">&quot;C&quot;</span>, row, <span class="st">&quot;: &quot;</span>, <span class="st">&quot;Latent Gaussians&quot;</span>))</span>
<span id="cb1-57"><a href="#cb1-57"></a>  <span class="kw">print</span>(x)</span>
<span id="cb1-58"><a href="#cb1-58"></a><span class="co"># print AUCs, note they are identical (for each row)  </span></span>
<span id="cb1-59"><a href="#cb1-59"></a><span class="kw">options</span>(<span class="dt">digits =</span> <span class="dv">9</span>)</span>
<span id="cb1-60"><a href="#cb1-60"></a>  <span class="kw">cat</span>(<span class="st">&quot;row =&quot;</span>, row, <span class="st">&quot;, seed =&quot;</span>, seed, <span class="st">&quot;, f =&quot;</span>, f, </span>
<span id="cb1-61"><a href="#cb1-61"></a>      <span class="st">&quot;</span><span class="ch">\n</span><span class="st">AUC of actual Gaussians =&quot;</span>, AUC1, </span>
<span id="cb1-62"><a href="#cb1-62"></a>      <span class="st">&quot;</span><span class="ch">\n</span><span class="st">AUC of latent Gaussians =&quot;</span>, AUC2, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb1-63"><a href="#cb1-63"></a>}</span>
<span id="cb1-64"><a href="#cb1-64"></a><span class="co">#&gt; row = 1 , seed = 10 , f = 0.1 </span></span>
<span id="cb1-65"><a href="#cb1-65"></a><span class="co">#&gt; AUC of actual Gaussians = 0.99308 </span></span>
<span id="cb1-66"><a href="#cb1-66"></a><span class="co">#&gt; AUC of latent Gaussians = 0.99308</span></span>
<span id="cb1-67"><a href="#cb1-67"></a><span class="co">#&gt; row = 2 , seed = 11 , f = 0.5 </span></span>
<span id="cb1-68"><a href="#cb1-68"></a><span class="co">#&gt; AUC of actual Gaussians = 0.993668889 </span></span>
<span id="cb1-69"><a href="#cb1-69"></a><span class="co">#&gt; AUC of latent Gaussians = 0.993668889</span></span>
<span id="cb1-70"><a href="#cb1-70"></a><span class="co">#&gt; row = 3 , seed = 12 , f = 0.9 </span></span>
<span id="cb1-71"><a href="#cb1-71"></a><span class="co">#&gt; AUC of actual Gaussians = 0.995041111 </span></span>
<span id="cb1-72"><a href="#cb1-72"></a><span class="co">#&gt; AUC of latent Gaussians = 0.995041111</span></span></code></pre></div>
<p><img src="06-binormal-model_files/figure-html/unnamed-chunk-4-1.png" width="33%" /><img src="06-binormal-model_files/figure-html/unnamed-chunk-4-2.png" width="33%" /><img src="06-binormal-model_files/figure-html/unnamed-chunk-4-3.png" width="33%" /><img src="06-binormal-model_files/figure-html/unnamed-chunk-4-4.png" width="33%" /><img src="06-binormal-model_files/figure-html/unnamed-chunk-4-5.png" width="33%" /><img src="06-binormal-model_files/figure-html/unnamed-chunk-4-6.png" width="33%" /><img src="06-binormal-model_files/figure-html/unnamed-chunk-4-7.png" width="33%" /><img src="06-binormal-model_files/figure-html/unnamed-chunk-4-8.png" width="33%" /><img src="06-binormal-model_files/figure-html/unnamed-chunk-4-9.png" width="33%" /></p>
<p><strong>Figure captions (A1 - C3):</strong> Illustrating the invariance of ROC analysis to arbitrary monotone transformations of the ratings. Each row contains 3 plots: labeled 1, 2 and 3. Each column contains 3 plots labeled A, B and C. So, for example, plot C2 refers to the second row and third column. The for-loop generates the plot one row at a time. Each of the latent Gaussian plots C1, C2 and C3 appears not binormal. However, using the inverse of the monotone transformations shown B1, B2 and B3, they can be transformed to the binormal model histograms A1, A2 and A3. Plot A1 shows the histogram of simulated ratings from a binormal model. Two peaks, one at 30 and the other at 55 are evident (by design, all ratings in this figure are in the range 0 to 100). Plot B1 shows the monotone transformation for <span class="math inline">\(f = 0.1\)</span>. Plot C1 shows the histogram of the transformed rating. The choice of <span class="math inline">\(f\)</span> leads to a transformed rating histogram that is peaked near the high end of the rating scale. For A1 and C1 the corresponding AUCs are identical (0.993080000). Plot A2 is for a different seed value, plot B2 is the transformation for <span class="math inline">\(f = 0.5\)</span> and now the transformed histogram is almost flat, plot C2. For plots A2 and C2 the corresponding AUCs are identical (0.993668889). Plot A3 is for a different seed value, B3 is the transformation for <span class="math inline">\(f = 0.9\)</span> and the transformed histogram C3 is peaked near the low end of the transformed rating scale. For plots A3 and (C3) the corresponding AUCs are identical (0.995041111).</p>
<p>The idea is to simulate continuous ratings data in the range 0 to 100 from a binormal model. <span class="math inline">\(K_1 = 900\)</span> non-diseased cases are sampled from a Gaussian centered at <span class="math inline">\(\mu_1\)</span> = 30 and standard deviation <span class="math inline">\(\sigma_1 = 7\)</span>. <span class="math inline">\(K_2 = 1000\)</span> diseased cases are sampled from a Gaussian centered at <span class="math inline">\(\mu_2\)</span> = 55 and standard deviation <span class="math inline">\(\sigma_2\)</span> = 7. The variable <span class="math inline">\(f\)</span>, which is in the range (0,1), controls the shape of the transformed distribution. If <span class="math inline">\(f\)</span> is small, the transformed distribution will be peaked towards 0 and if <span class="math inline">\(f\)</span> is unity, it will be peaked at 100. If <span class="math inline">\(f\)</span> equals 0.5, the transformed distribution is flat. Insight into the reason for this transformation is in <span class="citation">[@RN300]</span>, Chapter 7: it has to do with transformations of random variables. The transformation function, <span class="math inline">\(Y(Z)\)</span>, implements:</p>
<p><span class="math display">\[\begin{equation} 
Y\left ( Z \right )=\left [ \left ( 1-f \right )\Phi\left ( \frac{Z-\mu_1}{\sigma_1} \right )+f\Phi\left ( \frac{Z-\mu_2}{\sigma_2} \right ) \right ]100
(\#eq:BinModDemoMisconception)
\end{equation}\]</span></p>
<p>The multiplication by 100 ensures that the transformed variable is in the range 0 to 100 (if not, it is code-constrained to be). The code realizes the random samples, calculates the empirical AUC, displays the histogram of the true binormal samples, plots the transformation function, calculates the empirical AUC using the transformed samples, and plots the histogram of the transformed samples (the latent binormal).</p>
<ul>
<li>B1 shows the transformation for <span class="math inline">\(f = 0.1\)</span>. The steep initial rise of the curve has the effect of flattening the histogram of the transformed ratings at the low end of the rating scale, C1. Conversely, the flat nature of the curve near upper end of the rating range has the effect of causing the histogram of the transformed variable to peak in that range.</li>
<li>B2 shows the transformation for <span class="math inline">\(f = 0.5\)</span>. This time the latent rating histogram, C2, is almost flat over the entire range, definitely not visually binormal.</li>
<li>B3 shows the transformation for <span class="math inline">\(f = 0.9\)</span>. This time the transformed rating histogram, C3, is peaked at the low end of the transformed rating scale.</li>
<li>The output lists the values of the seed variable and the value of the shape parameter <span class="math inline">\(f\)</span>. <em>For each value of seed and the shape parameter, the AUCs of the actual Gaussians and the transformed variables are identical</em>.</li>
<li>The values of the parameters were chosen to best illustrate the true binormal nature of the plots A2 and A3. This has the effect of making the AUCs close to unity.</li>
</ul>
<p>The histograms in C1, C2 and C3 appear to be non-Gaussian. The corresponding non-diseased and diseased ratings will fail tests of normality. [Showing this is left as an exercise for the reader.] Nevertheless, they are latent Gaussians in the sense that the inverses of the transformations shown in B1, B2 and B3 will yield histograms that are strictly binormal, i.e., A1, A2 and A3. By appropriate changes to the monotone transformation function, the histograms shown in C1, C2 and C3 can be made to resemble a wide variety of shapes, for example, quasi-bimodal (don’t confuse bimodal with binormal) histograms.]</p>
<p><strong>Visual examination of the shape of the histograms of ratings, or standard tests for normality, yield little, if any, insight into whether the underlying binormal model assumptions are being violated.</strong></p>
</div>
<div id="expressions-for-sensitivity-and-specificity" class="section level3" number="1.2.3">
<h3 number="1.2.3"><span class="header-section-number">1.2.3</span> Expressions for sensitivity and specificity</h3>
<p>Let <span class="math inline">\(Z_t\)</span> denote the random z-sample for truth state <span class="math inline">\(t\)</span> (<span class="math inline">\(t\)</span> = 1 for non-diseased and <span class="math inline">\(t\)</span> = 2 for diseased cases). Since the distribution of z-samples from disease-free cases is <span class="math inline">\(N(0,1)\)</span>, the expression for specificity, Chapter “Modeling Binary Paradigm”, Eqn. 3.13, applies. It is reproduced below:</p>
<p><span class="math display">\[\begin{equation} 
Sp\left ( \zeta \right )=P\left ( Z_1 &lt; \zeta \right )=\Phi\left ( \zeta \right )
(\#eq:BinModSp)
\end{equation}\]</span></p>
<p>To obtain an expression for sensitivity, consider that for truth state <span class="math inline">\(t = 2\)</span>, the random variable <span class="math inline">\(\frac{Z_2-\mu}{\sigma}\)</span> is distributed as <span class="math inline">\(N(0,1)\)</span>:</p>
<p><span class="math display">\[\begin{equation*} 
\frac{Z_2-\mu}{\sigma}\sim N\left ( 0,1 \right )
\end{equation*}\]</span></p>
<p>Sensitivity is <span class="math inline">\(P\left ( Z_2 &gt; \zeta \right )\)</span>, which implies, because <span class="math inline">\(\sigma\)</span> is positive (subtract <span class="math inline">\(\mu\)</span> from both sides of the “greater than” symbol and divide by <span class="math inline">\(\sigma\)</span>):</p>
<p><span class="math display">\[\begin{equation} 
Se\left ( \zeta | \mu, \sigma \right )= P\left ( Z_2 &gt; \zeta \right )=P\left ( \frac{Z_2-\mu}{\sigma} &gt; \frac{\zeta-\mu}{\sigma} \right )
(\#eq:BinModSe)
\end{equation}\]</span></p>
<p>The right-hand-side can be rewritten as follows:</p>
<p><span class="math display">\[\begin{equation*} 
Se\left ( \zeta | \mu, \sigma \right )= 1 - P\left ( \frac{Z_2-\mu}{\sigma} \leq  \frac{\zeta-\mu}{\sigma} \right )\\
=1-\Phi\left (  \frac{\zeta-\mu}{\sigma}\right )=\Phi\left (  \frac{\mu-\zeta}{\sigma}\right )
\end{equation*}\]</span></p>
<p>Summarizing, the formulae for the specificity and sensitivity for the binormal model are:</p>
<p><span class="math display">\[\begin{equation} 
Sp\left ( \zeta \right ) = \Phi\left ( \zeta \right )\\
Se\left ( \zeta | \mu, \sigma \right ) = \Phi\left (  \frac{\mu-\zeta}{\sigma}\right )
(\#eq:BinModSeSp)
\end{equation}\]</span></p>
<p>The coordinates of the operating point defined by <span class="math inline">\(\zeta\)</span> are given by:</p>
<p><span class="math display">\[\begin{equation} 
FPF\left ( \zeta \right ) = 1 - Sp\left ( \zeta \right ) = 1 - \Phi\left ( \zeta \right ) = \Phi\left ( -\zeta \right )
(\#eq:BinModFPF)
\end{equation}\]</span></p>
<p><span class="math display">\[\begin{equation} 
TPF\left ( \zeta | \mu, \sigma \right ) = \Phi\left ( \frac{\mu-\zeta}{\sigma} \right )
(\#eq:BinModTPF)
\end{equation}\]</span></p>
<p>These expressions allow calculation of the operating point for any <span class="math inline">\(\zeta\)</span>. An equation for a curve is usually expressed as <span class="math inline">\(y=f(x)\)</span>. An expression of this form for the ROC curve, i.e., the y coordinate (TPF) expressed as a function of the x coordinate (FPF), follows upon inversion of the expression for FPF, Eqn. @ref(eq:BinModFPF):</p>
<p><span class="math display">\[\begin{equation} 
\zeta = -\Phi^{-1}\left ( FPF \right )
(\#eq:BinModZeta)
\end{equation}\]</span></p>
<p>Substitution of Eqn. @ref(eq:BinModZeta) in Eqn. @ref(eq:BinModTPF) yields:</p>
<p><span class="math display">\[\begin{equation} 
TPF = \Phi\left ( \frac{\mu + \Phi^{-1}\left (FPF  \right )}{\sigma} \right )
(\#eq:BinModRocCurve1)
\end{equation}\]</span></p>
<p>This equation gives the dependence of TPF on FPF, i.e., the equation for the ROC curve. It will be put into standard notation next.</p>
</div>
<div id="binormal-model-in-standard-notation" class="section level3" number="1.2.4">
<h3 number="1.2.4"><span class="header-section-number">1.2.4</span> Binormal model in standard notation</h3>
<p>The following notation is widely used in the literature:</p>
<p><span class="math display">\[\begin{equation} 
a=\frac{\mu}{\sigma};b=\frac{1}{\sigma}
(\#eq:BinModabParameters)
\end{equation}\]</span></p>
<p>The reason for the <span class="math inline">\((a,b)\)</span> instead of the <span class="math inline">\((\mu,\sigma)\)</span> notation is that Dorfman and Alf assumed, in their seminal paper <span class="citation">[@RN1081]</span>, that the diseased distribution (signal distribution in signal detection theory) had unit variance, and the non-diseased distribution (noise) had standard deviation <span class="math inline">\(b\)</span> (<span class="math inline">\(b &gt; 0\)</span>) or variance <span class="math inline">\(b^2\)</span>, and that the separation of the two distributions was <span class="math inline">\(a\)</span>, see figure below. In this example: <span class="math inline">\(a = 1.11\)</span> and <span class="math inline">\(b = 0.556\)</span>, corresponding to <span class="math inline">\(\mu = 2\)</span> and <span class="math inline">\(\sigma = 1.8\)</span>. Dorfman and Alf’s fundamental contribution, namely estimating these parameters from ratings data, to be described below, led to the widespread usage of the <span class="math inline">\((a,b)\)</span> parameters estimated by their software (RSCORE), and its newer variants (e.g., RSCORE–II, ROCFIT and ROCKIT).</p>
<p>By dividing the z-samples by <span class="math inline">\(b\)</span>, the variance of the distribution labeled “Noise” becomes unity, its mean stays at zero, and the variance of the distribution labeled “Signal” becomes <span class="math inline">\(1/b\)</span>, and its mean becomes <span class="math inline">\(a/b\)</span>, as shown below. It illustrates that the inverses of Eqn. @ref(eq:BinModabParameters) are:</p>
<p><span class="math display">\[\begin{equation} 
\mu=\frac{a}{b};\sigma=\frac{1}{b}
(\#eq:BinModabParametersInv)
\end{equation}\]</span></p>
<p>Eqns. @ref(eq:BinModabParameters) and @ref(eq:BinModabParametersInv) allow conversion from one notation to another.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">grid.arrange</span>(p1,p2,<span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<div class="figure">
<img src="06-binormal-model_files/figure-html/BinMod-ab2-mu-sigma-1.png" alt="The left plot shows the definitions of the (a,b) parameters of the binormal model. In the right plot the x-axis has been rescaled so that the noise distribution has unit variance, thereby illustrations between (a,b) and the ($\mu,\sigma$) parameters." width="672" />
<p class="caption">
(#fig:BinMod-ab2-mu-sigma)The left plot shows the definitions of the (a,b) parameters of the binormal model. In the right plot the x-axis has been rescaled so that the noise distribution has unit variance, thereby illustrations between (a,b) and the (<span class="math inline">\(\mu,\sigma\)</span>) parameters.
</p>
</div>
</div>
<div id="properties-of-the-binormal-model-roc-curve" class="section level3" number="1.2.5">
<h3 number="1.2.5"><span class="header-section-number">1.2.5</span> Properties of the binormal model ROC curve</h3>
<p>Using the <span class="math inline">\((a,b)\)</span> notation, Eqn. @ref(eq:BinModRocCurve1) for the ROC curve reduces to:</p>
<p><span class="math display">\[\begin{equation} 
TPF = \Phi\left ( a+ b \Phi^{-1}\left (FPF  \right ) \right )
(\#eq:BinModRocCurve)
\end{equation}\]</span></p>
<p>Since <span class="math inline">\(\Phi^{-1}(FPF)\)</span> is an increasing function of its argument <span class="math inline">\(FPF\)</span>, and <span class="math inline">\(b &gt; 0\)</span>, the argument of the <span class="math inline">\(\Phi\)</span> function is an increasing function of <span class="math inline">\(FPF\)</span>. Since <span class="math inline">\(\Phi\)</span> is a monotonically increasing function of its argument, <span class="math inline">\(TPF\)</span> is a monotonically increasing function of <span class="math inline">\(FPF\)</span>. This is true regardless of the sign of <span class="math inline">\(a\)</span>. If <span class="math inline">\(FPF = 0\)</span>, then <span class="math inline">\(\Phi^{-1}(0) = -\infty\)</span> and <span class="math inline">\(TPF = 0\)</span>. If <span class="math inline">\(FPF = 1\)</span>, then <span class="math inline">\(\Phi^{-1}(1) = +\infty\)</span> and <span class="math inline">\(TPF = 1\)</span>. Regardless of the value of <span class="math inline">\(a\)</span>, as long as <span class="math inline">\(b \ge 0\)</span>, the ROC curve starts at (0,0) and increases monotonically ending at (1,1).</p>
<p>From Eqn. @ref(eq:BinModFPF) and Eqn. @ref(eq:BinModTPF), the expressions for <span class="math inline">\(FPF\)</span> and <span class="math inline">\(TPF\)</span> in terms of model parameters <span class="math inline">\((a,b)\)</span> are:</p>
<p><span class="math display">\[\begin{equation}
\left.
\begin{aligned}
FPF\left ( \zeta \right ) &amp;= \Phi\left ( -\zeta \right )\\
TPF &amp;= \Phi\left ( a - b \zeta \right )
\end{aligned}
\right \}
(\#eq:BinModFPF-TPF-ab)
\end{equation}\]</span></p>
</div>
<div id="density-functions-pdfs-of-the-binormal-model" class="section level3" number="1.2.6">
<h3 number="1.2.6"><span class="header-section-number">1.2.6</span> Density functions (pdfs) of the binormal model</h3>
<p>According to Eqn. @ref(eq:BinModZSampling1) the probability that a z-sample is smaller than a specified threshold <span class="math inline">\(\zeta\)</span>, i.e., the CDF function, is:</p>
<p><span class="math display">\[\begin{equation*} 
P\left ( Z \le \zeta \mid  Z\sim N\left ( 0,1 \right ) \right ) = 1-FPF\left ( \zeta \right ) = \Phi \left ( \zeta  \right )
\end{equation*}\]</span></p>
<p><span class="math display">\[\begin{equation*} 
P\left ( Z \le \zeta \mid  Z\sim N\left ( \mu,\sigma^2 \right ) \right ) = 1-TPF\left ( \zeta \right ) = \Phi \left ( \frac{\zeta - \mu}{\sigma}  \right )
\end{equation*}\]</span></p>
<p>Since the <em>pdf</em> is the derivative of the corresponding CDF function, it follows that (the subscripts N and D denote non-diseased and diseased cases, respectively):</p>
<p><span class="math display">\[\begin{equation*} 
pdf_N\left ( \zeta \right ) = \frac{\partial \Phi\left ( \zeta \right )}{\partial \zeta} = \phi\left ( \zeta \right ) \equiv \frac{1}{\sqrt{2 \pi}}\exp\left ( -\frac{\zeta^2}{2} \right )
\end{equation*}\]</span></p>
<p><span class="math display">\[\begin{equation*} 
pdf_D\left ( \zeta \right ) = \frac{\partial \Phi\left ( \frac{\zeta - \mu}{\sigma} \right )}{\partial \zeta} = \frac{1}{\sigma} \phi\left ( \frac{\zeta - \mu}{\sigma} \right ) \equiv \frac{1}{\sqrt{2 \pi}\sigma}\exp\left ( -\frac{\left (\zeta-\mu  \right )^2}{2\sigma} \right )
\end{equation*}\]</span></p>
<p>The second equation can be written in <span class="math inline">\((a,b)\)</span> notation as:</p>
<p><span class="math display">\[\begin{equation*} 
pdf_D\left ( \zeta \right ) = b\phi\left ( b\zeta-a \right ) = \frac{b}{\sqrt{2 \pi}}\exp\left ( -\frac{\left (b\zeta - a \right )^2}{2} \right )
\end{equation*}\]</span></p>
</div>
</div>
<div id="fitting-an-roc-curve-to-data-points" class="section level2" number="1.3">
<h2 number="1.3"><span class="header-section-number">1.3</span> Fitting an ROC curve to data points</h2>
<div id="a-java-fitted-roc-curve" class="section level3" number="1.3.1">
<h3 number="1.3.1"><span class="header-section-number">1.3.1</span> A JAVA fitted ROC curve</h3>
<p>This section, described in the physical book, has been abbreviated to the <a href="http://www.rad.jhmi.edu/jeng/javarad/roc/JROCFITi.html">relevant website</a>.</p>
</div>
<div id="a-simplistic-straight-line-fit-to-the-roc-curve" class="section level3" number="1.3.2">
<h3 number="1.3.2"><span class="header-section-number">1.3.2</span> A simplistic straight line fit to the ROC curve</h3>
<p>To be described next is a method for fitting data such as in Table @ref(tab:ratingsParadigmExampleTable) to the binormal model, i.e., determining the parameters <span class="math inline">\((a,b)\)</span> and the thresholds <span class="math inline">\(\zeta_r , \quad r = 1, 2, ..., R-1\)</span>, to best fit, in some to-be-defined sense, the observed cell counts. The most common method uses an algorithm called maximum likelihood. But before getting to that, I describe the least-square method, which is conceptually simpler, but not really applicable, as will be explained shortly.</p>
<div id="least-squares-estimation" class="section level4" number="1.3.2.1">
<h4 number="1.3.2.1"><span class="header-section-number">1.3.2.1</span> Least-squares estimation</h4>
<p>By applying the function <span class="math inline">\(\Phi^{-1}\)</span> to both sides of Eqn. @ref(eq:BinModRocCurve1), one gets (the “inverse” function cancels the “forward” function on the right hand side):</p>
<p><span class="math display">\[\begin{equation*} 
\Phi^{-1}\left ( TPF \right ) = a + b \Phi^{-1}\left ( FPF \right )
\end{equation*}\]</span></p>
<p>This suggests that a plot of <span class="math inline">\(y = \Phi^{-1}\left ( TPF \right )\)</span> vs. <span class="math inline">\(x=\Phi^{-1}\left ( FPF \right )\)</span> is expected to follow a straight line with slope <span class="math inline">\(b\)</span> and intercept <span class="math inline">\(a\)</span>. Fitting a straight line to such data is generally performed by the method of least-squares, a capability present in most software packages and spreadsheets. Alternatively, one can simply visually draw the best straight line that fits the points, memorably referred to <span class="citation">[@RN300]</span> as “chi-by-eye”. This was the way parameters of the binormal model were estimated prior to Dorfman and Alf’s work <span class="citation">[@RN1081]</span>. The least-squares method is a quantitative way of accomplishing the same aim. If <span class="math inline">\(\left ( x_t,y_t \right )\)</span> are the data points, one constructs <span class="math inline">\(S\)</span>, the sum of the squared deviations of the observed ordinates from the predicted values (since <span class="math inline">\(R\)</span> is the number of ratings bins, the summation runs over the <span class="math inline">\(R-1\)</span> operating points):</p>
<p><span class="math display">\[\begin{equation*} 
S  = \sum_{i=1}^{R-1}\left ( y_i - \left ( a + bx_i \right ) \right )^2
\end{equation*}\]</span></p>
<p>The idea is to minimize S with respect to the parameters <span class="math inline">\((a,b)\)</span>. One approach is to differentiate this with respect to <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span> and equate each resulting derivate expression to zero. This yields two equations in two unknowns, which are solved for <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span>. If the reader has never done this before, one should go through these steps at least once, but it would be smarter in future to use software that does all this. In <code>R</code> the least-squares fitting function is <code>lm(y~x)</code>, which in its simplest form fits a linear model <code>lm(y~x)</code> using the method of least-squares (in case you are wondering <code>lm</code> stands for linear model, a whole branch of statistics in itself; in this example one is using its simplest capability).</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># ML estimates of a and b (from Eng JAVA program)</span></span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="co"># a &lt;- 1.3204; b &lt;- 0.6075 </span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co"># # these are not used in program; just here for comparison</span></span>
<span id="cb3-4"><a href="#cb3-4"></a></span>
<span id="cb3-5"><a href="#cb3-5"></a>FPF &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.017</span>, <span class="fl">0.050</span>, <span class="fl">0.183</span>, <span class="fl">0.5</span>)  </span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="co"># this is from Table 6.11, last two rows</span></span>
<span id="cb3-7"><a href="#cb3-7"></a>TPF &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.440</span>, <span class="fl">0.680</span>, <span class="fl">0.780</span>, <span class="fl">0.900</span>)</span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="co"># ...do...</span></span>
<span id="cb3-9"><a href="#cb3-9"></a></span>
<span id="cb3-10"><a href="#cb3-10"></a>PhiInvFPF &lt;-<span class="st"> </span><span class="kw">qnorm</span>(FPF)</span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="co"># apply the PHI_INV function</span></span>
<span id="cb3-12"><a href="#cb3-12"></a>PhiInvTPF &lt;-<span class="st"> </span><span class="kw">qnorm</span>(TPF)</span>
<span id="cb3-13"><a href="#cb3-13"></a><span class="co"># ... do ... </span></span>
<span id="cb3-14"><a href="#cb3-14"></a></span>
<span id="cb3-15"><a href="#cb3-15"></a>fit &lt;-<span class="st"> </span><span class="kw">lm</span>(PhiInvTPF<span class="op">~</span>PhiInvFPF)</span>
<span id="cb3-16"><a href="#cb3-16"></a><span class="kw">print</span>(fit)</span>
<span id="cb3-17"><a href="#cb3-17"></a><span class="co">#&gt; </span></span>
<span id="cb3-18"><a href="#cb3-18"></a><span class="co">#&gt; Call:</span></span>
<span id="cb3-19"><a href="#cb3-19"></a><span class="co">#&gt; lm(formula = PhiInvTPF ~ PhiInvFPF)</span></span>
<span id="cb3-20"><a href="#cb3-20"></a><span class="co">#&gt; </span></span>
<span id="cb3-21"><a href="#cb3-21"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb3-22"><a href="#cb3-22"></a><span class="co">#&gt; (Intercept)    PhiInvFPF  </span></span>
<span id="cb3-23"><a href="#cb3-23"></a><span class="co">#&gt;    1.328844     0.630746</span></span></code></pre></div>
<div class="figure">
<img src="06-binormal-model_files/figure-html/BinModStraightLineFit-1.png" alt="The straight line fit method of estimating parameters of the fitting model." width="672" />
<p class="caption">
(#fig:BinModStraightLineFit)The straight line fit method of estimating parameters of the fitting model.
</p>
</div>
<p>Fig. @ref(fig:BinModStraightLineFit) shows operating points from Table @ref(tab:ratingsParadigmExampleTable), transformed by the <span class="math inline">\(\Phi^{-1}\)</span> function; the slope of the line is the least-squares estimate of the <span class="math inline">\(b\)</span> parameter and the intercept is the corresponding <span class="math inline">\(a\)</span> parameter of the binormal model.</p>
<p>The last line contains the least squares estimated values, <span class="math inline">\(a\)</span> = 1.3288 and <span class="math inline">\(b\)</span> = 0.6307. The corresponding maximum likelihood estimates of these parameters, as yielded by the Eng web code, Appendix B, are listed in line 4 of the main program: <span class="math inline">\(a\)</span> = 1.3204 and <span class="math inline">\(b\)</span> = 0.6075. The estimates appear to be close, particularly the estimate of <span class="math inline">\(a\)</span> , but there are a few things wrong with the least-squares approach. First, the method of least squares assumes that the data points are independent. Because of the manner in which they are constructed, namely by cumulating points, the independence assumption is not valid for ROC operating points. Cumulating the 4 and 5 responses constrains the resulting operating point to be above and to the right of the point obtained by cumulating the 5 responses only, so the data points are definitely not independent. Similarly, cumulating the 3, 4 and 5 responses constrains the resulting operating point to be above and to the right of the point obtained by cumulating the 4 and 5 responses, and so on. The second problem is the linear least-squares method assumes there is no error in measuring x; the only source of error that is accounted for is in the y-coordinate. In fact, both coordinates of an ROC operating point are subject to sampling error. Third, disregard of error in the x-direction is further implicit in the estimates of the thresholds, which according to Eqn. (6.2.19), is given by:</p>
<p><span class="math display">\[\begin{equation*} 
\zeta_r = - \Phi^{-1}\left ( FPF_r \right )
\end{equation*}\]</span></p>
<p>These are “rigid” estimates that assume no error in the FPF values. As was shown in Chapter @ref(binaryTask), 95% confidence intervals apply to these estimates.</p>
<p>A historical note: prior to computers and easy access to statistical functions the analyst had to use a special plotting paper, termed “double probability paper”, that converted probabilities into x and y distances using the inverse function.</p>
</div>
</div>
<div id="maximum-likelihood-estimation-mle" class="section level3" number="1.3.3">
<h3 number="1.3.3"><span class="header-section-number">1.3.3</span> Maximum likelihood estimation (MLE)</h3>
<p>The approach taken by Dorfman and Alf was to maximize the likelihood function instead of S. The likelihood function is the probability of the observed data given a set of parameter values, i.e.,</p>
<p><span class="math display">\[\begin{equation*} 
\text {L} \equiv P\left ( data \mid \text {parameters} \right )
\end{equation*}\]</span></p>
<p>Generally “data” is suppressed, so likelihood is a function of the parameters; but “data” is always implicit. With reference to Fig. 6.1, the probability of a non-diseased case yielding a count in the 2nd bin equals the area under the curve labeled “Noise” bounded by the vertical lines at <span class="math inline">\(\zeta_1\)</span> and <span class="math inline">\(\zeta_2\)</span>. In general, the probability of a non-diseased case yielding a count in the <span class="math inline">\(r^\text{th}\)</span> bin equals the area under the curve labeled “Noise” bounded by the vertical lines at <span class="math inline">\(\zeta_{r-1}\)</span> and <span class="math inline">\(\zeta_r\)</span>. Since the area to the left of a threshold is the CDF corresponding to that threshold, the required probability is <span class="math inline">\(\Phi\left ( \zeta_r \right ) - \Phi\left ( \zeta_{r-1} \right )\)</span>; we are simply subtracting two expressions for specificity, Eqn. (6.2.5).</p>
<p><span class="math display">\[\begin{equation*} 
\text {count in non-diseased bin } r = \Phi\left ( \zeta_r \right ) - \Phi\left ( \zeta_{r-1} \right )
\end{equation*}\]</span></p>
<p>Similarly, the probability of a diseased case yielding a count in the rth bin equals the area under the curve labeled “Signal” bounded by the vertical lines at <span class="math inline">\(\zeta_{r-1}\)</span> and <span class="math inline">\(\zeta_r\)</span>. The area under the diseased distribution to the left of threshold <span class="math inline">\(\zeta_r\)</span> is the <span class="math inline">\(1 - TPF\)</span> at that threshold:</p>
<p><span class="math display">\[\begin{equation*} 
1 - \Phi\left ( \frac{\mu-\zeta_r}{\sigma} \right ) = \Phi\left ( \frac{\zeta_r - \mu}{\sigma} \right )
\end{equation*}\]</span></p>
<p>The area between the two thresholds is:</p>
<p><span class="math display">\[\begin{align*} 
P\left ( \text{count in diseased bin }r \right ) &amp;= \Phi\left ( \frac{\zeta_r - \mu}{\sigma} \right ) - \Phi\left ( \frac{\zeta_{r-1} - \mu}{\sigma} \right ) \\
&amp;= \Phi\left ( b\zeta_r-a \right ) - \Phi\left ( b\zeta_{r-1}-a \right )
\end{align*}\]</span></p>
<p>Let <span class="math inline">\(K_{1r}\)</span> denote the number of non-diseased cases in the rth bin, and <span class="math inline">\(K_{2r}\)</span> denotes the number of diseased cases in the rth bin. Consider the number of counts <span class="math inline">\(K_{1r}\)</span> in non-diseased case bin <span class="math inline">\(r\)</span>. Since the probability of each count is <span class="math inline">\(\Phi\left ( \zeta_{r+1} \right ) - \Phi\left ( \zeta_r \right )\)</span>, the probability of the observed number of counts, assuming the counts are independent, is <span class="math inline">\({\left(\Phi\left ( \zeta_{r+1} \right ) - \Phi\left ( \zeta_r \right ) \right )}^{K_{1r}}\)</span>. Similarly, the probability of observing counts in diseased case bin <span class="math inline">\(r\)</span> is <span class="math inline">\({\left (\Phi\left ( b\zeta_{r+1}-a \right ) - \Phi\left ( b\zeta_r-a \right ) \right )}^{K_{2r}}\)</span>, subject to the same independence assumption. The probability of simultaneously observing <span class="math inline">\(K_{1r}\)</span> counts in non-diseased case bin r and <span class="math inline">\(K_{2r}\)</span> counts in diseased case bin <span class="math inline">\(r\)</span> is the product of these individual probabilities (again, an independence assumption is being used):</p>
<p><span class="math display">\[\begin{equation*} 
\left (\Phi\left ( \zeta_{r+1}  \right ) - \Phi\left ( \zeta_r  \right )  \right )^{K_{1r}} \left (\Phi\left ( b\zeta_{r+1}-a  \right ) - \Phi\left ( b\zeta_r-a  \right )  \right )^{K_{2r}}
\end{equation*}\]</span></p>
<p>Similar expressions apply for all integer values of <span class="math inline">\(r\)</span> ranging from <span class="math inline">\(1,2,...,R\)</span>. Therefore the probability of observing the entire data set is the product of expressions like Eqn. (6.4.5), over all values of <span class="math inline">\(r\)</span>:</p>
<p><span class="math display">\[\begin{equation} 
\prod_{r=1}^{R}\left [\left (\Phi\left ( \zeta_{r+1}  \right ) - \Phi\left ( \zeta_r  \right )  \right )^{K_{1r}} \left (\Phi\left ( b\zeta_{r+1}-a  \right ) - \Phi\left ( b\zeta_r-a  \right )  \right )^{K_{2r}}  \right ]
(\#eq:BinModProductProb)
\end{equation}\]</span></p>
<p>We are almost there. A specific combination of <span class="math inline">\(K_{11},K_{12},...,K_{1R}\)</span> counts from <span class="math inline">\(K_1\)</span> non-diseased cases and counts <span class="math inline">\(K_{21},K_{22},...,K_{2R}\)</span> from <span class="math inline">\(K_2\)</span> diseased cases can occur the following number of times (given by the multinomial factor shown below):</p>
<p><span class="math display">\[\begin{equation} 
\frac{K_1!}{\prod_{r=1}^{R}K_{1r}!}\frac{K_2!}{\prod_{r=1}^{R}K_{2r}!}
(\#eq:BinModCombFactor)
\end{equation}\]</span></p>
<p>The likelihood function is the product of Eqn. @ref(eq:BinModProductProb) and Eqn. @ref(eq:BinModCombFactor):</p>
<p><span class="math display">\[\begin{equation} 
\begin{split}
L\left ( a,b,\overrightarrow{\zeta} \right ) &amp;= \left (\frac{K_1!}{\prod_{r=1}^{R}K_{1r}!}\frac{K_2!}{\prod_{r=1}^{R}K_{2r}!}  \right ) \times \\
&amp;\quad\prod_{r=1}^{R}\left [\left (\Phi\left ( \zeta_{r+1}  \right ) - \Phi\left ( \zeta_r  \right )  \right )^{K_{1r}} \left (\Phi\left ( b\zeta_{r+1}-a  \right ) - \Phi\left ( b\zeta_r-a  \right )  \right )^{K_{2r}}  \right ]
\end{split}
(\#eq:BinModLikelihood)
\end{equation}\]</span></p>
<p>The left hand side of Eqn. @ref(eq:BinModLikelihood) shows explicitly the dependence of the likelihood function on the parameters of the model, namely <span class="math inline">\(a,b,\overrightarrow{\zeta}\)</span>, where the vector of thresholds <span class="math inline">\(\overrightarrow{\zeta}\)</span> is a compact notation for the set of thresholds <span class="math inline">\(\zeta_1,\zeta_2,...,\zeta_R\)</span>, (note that since <span class="math inline">\(\zeta_0 = -\infty\)</span>, and <span class="math inline">\(\zeta_R = +\infty\)</span>, only <span class="math inline">\(R-1\)</span> free threshold parameters are involved, and the total number of free parameters in the model is <span class="math inline">\(R+1\)</span>). For example, for a 5-rating ROC study, the total number of free parameters is 6, i.e., <span class="math inline">\(a\)</span>, <span class="math inline">\(b\)</span> and 4 thresholds <span class="math inline">\(\zeta_1,\zeta_2,\zeta_3,\zeta_4\)</span>.</p>
<p>Eqn. @ref(eq:BinModLikelihood) is forbidding but here comes a simplification. The difference of probabilities such as <span class="math inline">\(\Phi\left ( \zeta_r \right )-\Phi\left ( \zeta_{r-1} \right )\)</span> is guaranteed to be positive and less than one [the <span class="math inline">\(\Phi\)</span> function is a probability, i.e., in the range 0 to 1, and since <span class="math inline">\(\zeta_r\)</span> is greater than <span class="math inline">\(\zeta_{r-1}\)</span>, the difference is positive and less than one]. When the difference is raised to the power of <span class="math inline">\(K_{1r}\)</span> (a non-negative integer) a very small number can result. Multiplication of all these small numbers may result in an even smaller number, which may be too small to be represented as a floating-point value, especially as the number of counts increases. To prevent this we resort to a trick. Instead of maximizing the likelihood function <span class="math inline">\(L\left ( a,b,\overrightarrow{\zeta} \right )\)</span> we choose to maximize the logarithm of the likelihood function (the base of the logarithm is immaterial). The logarithm of the likelihood function is:</p>
<p><span class="math display">\[\begin{equation} 
LL\left ( a,b,\overrightarrow{\zeta} \right )=\log \left ( L\left ( a,b,\overrightarrow{\zeta} \right ) \right )
(\#eq:BinModLogLikelihood)
\end{equation}\]</span></p>
<p>Since the logarithm is a monotonically increasing function of its argument, maximizing the logarithm of the likelihood function is equivalent to maximizing the likelihood function. Taking the logarithm converts the product symbols in Eqn. (6.4.8) to summations, so instead of multiplying small numbers one is adding them, thereby avoiding underflow errors. Another simplification is that one can ignore the logarithm of the multinomial factor involving the factorials, because these do not depend on the parameters of the model. Putting all this together, we get the following expression for the logarithm of the likelihood function:</p>
<p><span class="math display">\[\begin{equation} 
\begin{split}
LL\left ( a,b,\overrightarrow{\zeta} \right ) \propto&amp; \sum_{r=1}^{R} K_{1r}\log \left ( \Phi\left ( \zeta_{r+1} \right ) - \Phi\left ( \zeta_r \right ) \right ) \\
&amp;+ \sum_{r=1}^{R} K_{2r}\log \left ( \Phi\left (b \zeta_{r+1} - a \right ) - \Phi\left ( b \zeta_r - a \right ) \right ) 
\end{split}
(\#eq:BinModLL)
\end{equation}\]</span></p>
<p>The left hand side of Eqn. @ref(eq:BinModLL) is a function of the model parameters <span class="math inline">\(a,b,\overrightarrow{\zeta}\)</span> and the observed data, the latter being the counts contained in the vectors <span class="math inline">\(\overrightarrow{K_1}\)</span> and <span class="math inline">\(\overrightarrow{K_2}\)</span>, where the vector notation is used as a compact form for the counts <span class="math inline">\(K_{11},K_{12},...,K_{1R}\)</span> and <span class="math inline">\(K_{21},K_{22},...,K_{2R}\)</span>, respectively. The right hand side of Eqn. @ref(eq:BinModLL) is monotonically related to the probability of observing the data given the model parameters <span class="math inline">\(a,b,\overrightarrow{\zeta}\)</span>. If the choice of model parameters is poor, then the probability of observing the data will be small and log likelihood will be small. With a better choice of model parameters the probability and log likelihood will increase. With optimal choice of model parameters the probability and log likelihood will be maximized, and the corresponding optimal values of the model parameters are called maximum likelihood estimates (MLEs). These are the estimates produced by the programs RSCORE and ROCFIT.</p>
</div>
<div id="code-implementing-mle" class="section level3" number="1.3.4">
<h3 number="1.3.4"><span class="header-section-number">1.3.4</span> Code implementing MLE</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a></span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co"># ML estimates of a and b (from Eng JAVA program)</span></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="co"># a &lt;- 1.3204; b &lt;- 0.6075 </span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co"># these are not used in program; just there for comparison</span></span>
<span id="cb4-5"><a href="#cb4-5"></a></span>
<span id="cb4-6"><a href="#cb4-6"></a>K1t &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">30</span>, <span class="dv">19</span>, <span class="dv">8</span>, <span class="dv">2</span>, <span class="dv">1</span>)</span>
<span id="cb4-7"><a href="#cb4-7"></a>K2t &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">5</span>,  <span class="dv">6</span>, <span class="dv">5</span>, <span class="dv">12</span>, <span class="dv">22</span>)</span>
<span id="cb4-8"><a href="#cb4-8"></a>dataset &lt;-<span class="st"> </span><span class="kw">Df2RJafrocDataset</span>(K1t, K2t, <span class="dt">InputIsCountsTable =</span> <span class="ot">TRUE</span>)</span>
<span id="cb4-9"><a href="#cb4-9"></a>retFit &lt;-<span class="st"> </span><span class="kw">FitBinormalRoc</span>(dataset)</span>
<span id="cb4-10"><a href="#cb4-10"></a>retFit[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span>
<span id="cb4-11"><a href="#cb4-11"></a><span class="co">#&gt; $a</span></span>
<span id="cb4-12"><a href="#cb4-12"></a><span class="co">#&gt; [1] 1.32045261</span></span>
<span id="cb4-13"><a href="#cb4-13"></a><span class="co">#&gt; </span></span>
<span id="cb4-14"><a href="#cb4-14"></a><span class="co">#&gt; $b</span></span>
<span id="cb4-15"><a href="#cb4-15"></a><span class="co">#&gt; [1] 0.607492932</span></span>
<span id="cb4-16"><a href="#cb4-16"></a><span class="co">#&gt; </span></span>
<span id="cb4-17"><a href="#cb4-17"></a><span class="co">#&gt; $zetas</span></span>
<span id="cb4-18"><a href="#cb4-18"></a><span class="co">#&gt;      zetaFwd1      zetaFwd2      zetaFwd3      zetaFwd4 </span></span>
<span id="cb4-19"><a href="#cb4-19"></a><span class="co">#&gt; 0.00768054675 0.89627306763 1.51564784976 2.39672209865 </span></span>
<span id="cb4-20"><a href="#cb4-20"></a><span class="co">#&gt; </span></span>
<span id="cb4-21"><a href="#cb4-21"></a><span class="co">#&gt; $AUC</span></span>
<span id="cb4-22"><a href="#cb4-22"></a><span class="co">#&gt; [1] 0.870452157</span></span>
<span id="cb4-23"><a href="#cb4-23"></a><span class="co">#&gt; </span></span>
<span id="cb4-24"><a href="#cb4-24"></a><span class="co">#&gt; $StdAUC</span></span>
<span id="cb4-25"><a href="#cb4-25"></a><span class="co">#&gt;              [,1]</span></span>
<span id="cb4-26"><a href="#cb4-26"></a><span class="co">#&gt; [1,] 0.0379042262</span></span>
<span id="cb4-27"><a href="#cb4-27"></a><span class="kw">print</span>(retFit<span class="op">$</span>fittedPlot)</span></code></pre></div>
<p><img src="06-binormal-model_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>Note the usage of the <code>RJafroc</code> package <span class="citation">[@R-RJafroc]</span>. Specifically, the function <code>FitBinormalRoc</code>. The ratings table is converted to an <code>RJafroc</code> dataset object, followed by application of the fitting function. The results, contained in <code>retFit</code> should be compared to those obtained from the <a href="http://www.rad.jhmi.edu/jeng/javarad/roc/JROCFITi.html">website implementation of ROCFIT</a>.</p>
</div>
<div id="validating-the-fitting-model" class="section level3" number="1.3.5">
<h3 number="1.3.5"><span class="header-section-number">1.3.5</span> Validating the fitting model</h3>
<p>The above ROC curve is a good visual fit to the observed operating points. Quantification of the validity of the fitting model is accomplished by calculating the Pearson goodness-of-fit test <span class="citation">[@RN2656]</span>, also known as the chi-square test, which uses the statistic defined by <span class="citation">[@RN1492]</span>:</p>
<p><span class="math display">\[\begin{equation} 
C^2=\sum_{t=1}^{2}\sum_{r=1}^{R}\frac{\left (K_{tr}-\left \langle K_{tr} \right \rangle  \right )^2}{\left \langle K_{tr} \right \rangle}\\
K_{tr} \geq 5
(\#eq:BinModGoodnessFit)
\end{equation}\]</span></p>
<p>The expected values are given by:</p>
<p><span class="math display">\[\begin{equation}
\begin{split}
\left \langle K_{1r} \right \rangle &amp;=K_1\left ( \Phi\left ( \zeta_{r+1} \right ) - \Phi\left ( \zeta_r \right )  \right ) \\
\left \langle K_{2r} \right \rangle &amp;=K_2\left ( \Phi\left ( a\zeta_{r+1}-b \right ) - \Phi\left ( a\zeta_r - b\right )  \right )
\end{split}
(\#eq:BinModGoodnessFitExpVals)
\end{equation}\]</span></p>
<p>These expressions should make sense: the difference between the two CDF functions is the probability of a count in the specified bin, and multiplication by the total number of relevant cases should yield the expected counts (a non-integer).</p>
<p>It can be shown that under the null hypothesis that the assumed probability distribution functions for the counts equals the true probability distributions, i.e., the model is valid, the statistic <span class="math inline">\(C^2\)</span> is distributed as:</p>
<p><span class="math display">\[\begin{equation} 
C^2\sim \chi_{df}^{2}
(\#eq:BinModGoodnessFitDistr)
\end{equation}\]</span></p>
<p>Here <span class="math inline">\(C^2\sim \chi_{df}^{2}\)</span> is the chi-square distribution with degrees of freedom <em>df</em> defined by:</p>
<p><span class="math display">\[\begin{equation} 
df=\left ( R-1 \right )+\left ( R-1 \right )-\left (2+ R-1 \right )=\left ( R-3 \right )
(\#eq:BinModGoodnessFitdf)
\end{equation}\]</span></p>
<p>The right hand side of the above equation has been written in an expansive form to illustrate the general rule: for <span class="math inline">\(R\)</span> non-diseased cells in the ratings table, the degree of freedom is <span class="math inline">\(R-1\)</span>: this is because when all but one cells are specified, the last is determined, because they must sum to <span class="math inline">\(K_1\)</span> . Similarly, the degree of freedom for the diseased cells is also <span class="math inline">\(R-1\)</span>. Last, we need to subtract the number of free parameters in the model, which is <span class="math inline">\((2+R-1)\)</span>, i.e., the <span class="math inline">\(a,b\)</span> parameters and the <span class="math inline">\(R-1\)</span> thresholds. It is evident that if <span class="math inline">\(R = 3\)</span> then <span class="math inline">\(df = 0\)</span>. In this situation, there are only two non-trivial operating points and the straight-line fit shown will pass through both of them. With two basic parameters, fitting two points is trivial, and goodness of fit cannot be calculated.</p>
<p>Under the null hypothesis (i.e., model is valid) <span class="math inline">\(C^2\)</span> is distributed as <span class="math inline">\(\chi_{df}^{2}\)</span>. Therefore, one computes the probability that this statistic is larger than the observed value, called the <em>p-value</em>. If this probability is very small, that means that the deviations of the observed values of the cell counts from the expected values are so large that it is unlikely that the model is correct. The degree of unlikeliness is quantified by the p-value. Poor fits lead to small p values.</p>
<p>At the 5% significance level, one concludes that the fit is not good if <span class="math inline">\(p &lt; 0.05\)</span>. In practice one occasionally accepts smaller values of <span class="math inline">\(p\)</span>, <span class="math inline">\(p &gt; 0.001\)</span> before completely abandoning a model. It is known that adoption of a stricter criterion, e.g., <span class="math inline">\(p &gt; 0.05\)</span>, can occasionally lead to rejection of a retrospectively valid model <span class="citation">[@RN300]</span>.</p>
</div>
<div id="estimating-the-covariance-matrix" class="section level3" number="1.3.6">
<h3 number="1.3.6"><span class="header-section-number">1.3.6</span> Estimating the covariance matrix</h3>
<p>TBA See book chapter 6.4.3. This is implemented in RJafroc.</p>
</div>
<div id="estimating-the-variance-of-az" class="section level3" number="1.3.7">
<h3 number="1.3.7"><span class="header-section-number">1.3.7</span> Estimating the variance of Az</h3>
<p>TBA See book chapter 6.4.4. This is implemented in RJafroc.</p>
</div>
<div id="single-fom-derived-from-roc-curve" class="section level3" number="1.3.8">
<h3 number="1.3.8"><span class="header-section-number">1.3.8</span> Single FOM derived from ROC curve</h3>
<p>Sensitivity and specificity are <em>dual</em> measures of overall performance. It is hard to unambiguosly compare two systems usng dual measures. What if sensitivity is higher for one system but specificity is higher for another. This is, of course, a consequence of sensitivity/specificity depending on the position of the operating point on the ROC curve. Desirable is a <em>single</em> measure of performance that takes into account performance over the entire ROC curve. Two commonly used measures are the binormla model predicted area <span class="math inline">\(A_z\)</span> under the ROC curve, and the <span class="math inline">\(d&#39;\)</span> index.</p>
<p>TBA (Book) Appendix 6.A derives the formula for the partial area under the unequal-variance binormal model. A special case of this formula is the area under the whole ROC curve, reproduced below using both parameterizations of the model:</p>
<p><span class="math display">\[\begin{equation} 
A_z=\Phi\left ( \frac{a}{\sqrt{1+b^2}} \right )=\Phi\left ( \frac{\mu}{\sqrt{1+\sigma^2}} \right )
(\#eq:BinModab2Az)
\end{equation}\]</span></p>
<p>The binormal fitted AUC increases as <span class="math inline">\(a\)</span> increases or as <span class="math inline">\(b\)</span> decreases. Equivalently, it increases as <span class="math inline">\(\mu\)</span> increases or as <span class="math inline">\(\sigma\)</span> decreases. An equivalent <span class="math inline">\(d&#39;\)</span> parameter is defined as the separation of two unit-variance normal distributions yielding the same AUC as that predicted by the <span class="math inline">\((a,b)\)</span> parameter binormal model. It is defined by:</p>
<p><span class="math display">\[\begin{equation} 
d&#39;=\sqrt{2}\Phi^{-1}\left ( A_z \right )
(\#eq:BinModab2dprime)
\end{equation}\]</span></p>
</div>
</div>
<div id="BinMod-Discussion" class="section level2" number="1.4">
<h2 number="1.4"><span class="header-section-number">1.4</span> Discussion</h2>
<p>The binormal model is historically very important and the contribution by Dorfman and Alf <span class="citation">[@RN1081]</span> was seminal. Prior to their work, there was no valid way of estimating AUC from observed ratings counts. Their work and a key paper <span class="citation">[@RN1487]</span> accelerated research using ROC methods. The number of publications using their algorithm, and the more modern versions developed by Metz and colleagues, is probably well in excess of 500. Because of its key role, the author has endeavored to take out some of the mystery about how the binormal model parameters are estimated. In particular, a common misunderstanding that the binormal model assumptions are violated by real datasets, when in fact it is quite robust to apparent deviations from normality, is addressed.</p>
<p>A good understanding of this chapter should enable the reader to better understand alternative ROC models, discussed later.</p>
<p>It has been stated that the <code>b</code>-parameter of the binormal model is generally observed to be less than one, consistent with the diseased distribution being wider than the non-diseased one. The ROC literature is largely silent on the reason for this finding. One reason, namely location uncertainty, is presented in Chapter “Predictions of the RSM”, where RSM stands for Radiological Search Model. Basically, if the location of the lesion is unknown, then z-samples from diseased cases can be of two types, samples from the correct lesion location, or samples from other non-lesion locations. The resulting mixture distribution will then appear to have larger variance than the corresponding samples from non-diseased cases. This type of mixing need not be restricted to location uncertainty. Even is location is known, if the lesions are non-homogenous (e.g., they contain a range of contrasts) then a similar mixture-distribution induced broadening is expected. The contaminated binormal model (CBM) - see Chapter TBA - also predicts that the diseased distribution is wider than the non-diseased one.</p>
<p>The fact that the <code>b</code>-parameter is less than unity implies that the predicted ROC curve is improper, meaning its slope is not monotone decreasing as the operating point moves up the curve. The result is that a portion of the curve, near (1,1) that crosses the chance-diagonal and hooks upward approaching (1,1) with infinite slope. Ways of fitting proper ROC curves are described in Chapter “Other proper ROC models”. Usually the hook is not readily visible, which has been used as an excuse to ignore the problem. For example, in Fig. 6.4, one would have to “zoom-in” on the upper right corner to see it, but the reader should make no mistake about it, the hook is there as .</p>
<p>A recent example is Fig. 1 in the publication resulting from the Digital Mammographic Imaging Screening Trial (DMIST) clinical trial <span class="citation">[@RN1784]</span> involving 49,528 asymptomatic women from 33 clinical sites and involving 153 radiologists, where each of the film modality ROC plots crosses the chance diagonal and hooks upwards to (1,1), which as is known, results anytime <span class="math inline">\(b &lt;1\)</span>.</p>
<p>The unphysical nature of the hook (predicting worse than chance-level performance for supposedly expert readers) is not the only reason for seeking alternate ROC models. The binormal model is susceptible to degeneracy problems. If the dataset does not provide any interior operating points (i.e., all observed points lie on the axes defined by FPF = 0 or TPF = 1) then the model fits these points with b = 0. The resulting straight-line segment fits do not make physical sense. These problems are addressed by the contaminated binormal model16 to be discussed in Chapter “Other proper ROC models”. The first paper in the series has particularly readable accounts of data degeneracy.</p>
<p>To this day the binormal model is widely used to fit ROC datasets. In spite of its limitations, the binormal model has been very useful in bringing a level of quantification to this field that did not exist prior to <span class="citation">[@RN1081]</span>.</p>
</div>
<div id="BinMod-references" class="section level2" number="1.5">
<h2 number="1.5"><span class="header-section-number">1.5</span> References</h2>
</div>
</div>
<!--bookdown:body:end-->
            </section>

          </div>
        </div>
      </div>
<!--bookdown:link_prev-->
<!--bookdown:link_next-->
    </div>
  </div>
<!--bookdown:config-->

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
